File: src/main/java/ru/grabovsky/recordkeeping/models/entities/Authority.java
package ru.grabovsky.recordkeeping.models.entities;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;

import javax.persistence.*;
import java.util.Objects;
import java.util.Set;

/**
 * Entity for authorities table
 *
 * @author GrabovskyAlexey
 */
@Entity
@Table(name = "authorities")
@Getter
@Setter
@ToString
public class Authority {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "authority", nullable = false)
    private String authority;

    @Column(name = "description", nullable = false)
    private String description;

    @ManyToMany(mappedBy = "authorities")
    @ToString.Exclude
    private Set<Role> roles;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Authority authority = (Authority) o;
        return id != null && Objects.equals(id, authority.id);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: core/src/main/kotlin/ru/grabovsky/recordkeeping/core/entity/db/Authority.kt
package ru.grabovsky.recordkeeping.core.entity.db

import jakarta.persistence.*
import ru.grabovsky.recordkeeping.api.types.AuthorityTypes

/**
 * Entity for authorities table
 *
 * @author GrabovskyAlexey
 */
@Entity
@Table(name = "authorities")
class Authority(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    override var id: Long? = null,
    @Column(name = "description", nullable = false)
    var description: String,
    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    val type: AuthorityTypes
): BaseEntity(id) {

    @ManyToMany(mappedBy = "authorities")
    val roles: MutableSet<Role> = mutableSetOf()


    override fun toString(): String {
        return "Authority(id=$id, authority='$type', description='$description')"
    }

}